drivers/gpio/gpio_max149x6.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2024 Analog Devices Inc.
+ * Copyright (c) 2024 Baylibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define MAX149x6_READ 0
+#define MAX149x6_WRITE 1
+
+#define PRINT_ERR_BIT(bit1, bit2) if ((bit1) & (bit2)) LOG_ERR("[%s] %d", #bit1, bit1)
+#define PRINT_ERR(bit) if (bit) LOG_ERR("[DIAG] [%s] %d\n", #bit, bit)
+#define PRINT_INF(bit) LOG_INFO("[%s] %d\n", #bit, bit)
+#define LOG_DIAG(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
+
+/**
+ * @brief Compute the CRC5 value for an array of bytes when writing to MAX149X6
+ * @param data - array of data to encode
+ * @param encode - action to be performed - true(encode), false(decode)
+ * @return the resulted CRC5
+ */
+static uint8_t max149x6_crc(uint8_t *data, bool encode)
```

Comment:
Why is this implemented in the header file?

Reply:
because it is shared from MAX14906 and MAX14916

Reply:
How about to to rename it as .c instead .h to it be more readable.

Reply:
I am not sure if will be more readable. I prefer to keep it with .h to avoid confusion if max149x6 is separate driver.

Reply:
It is ok, thanks.